In [10]:
import pandas as pd
import os
import numpy as np

df = pd.read_json('../Data/Serie_A_2014_squads.json')

df = df.T
df

0  \
1    {'Home': {'Squad': ['141698', '137644', '14474...   
2    {'Home': {'Squad': ['137990', '172996', '16795...   
3    {'Home': {'Squad': ['107639', '161940', '45966...   
4    {'Home': {'Squad': ['142720', '185679', '16194...   
5    {'Home': {'Squad': ['293428', '322380', '18260...   
..                                                 ...   
376  {'Home': {'Squad': ['180213', '174360', '15728...   
377  {'Home': {'Squad': ['511533', '169529', '29338...   
378  {'Home': {'Squad': ['168351', '165065', '16928...   
379  {'Home': {'Squad': ['291584', '171788', '16103...   
380  {'Home': {'Squad': ['155703', '176512', '30325...   

                                                     1  \
1    {'Home': {'Squad': ['141698', '137644', '14474...   
2    {'Home': {'Squad': ['137990', '172996', '16795...   
3    {'Home': {'Squad': ['107639', '161940', '45966...   
4    {'Home': {'Squad': ['142720', '185679', '16194...   
5    {'Home': {'Squad': ['293428', '322380', '18260...   
..                                                 ...   
376  {'Home': {'Squad': ['180213', '174360', '15728...   
377  {'Home': {'Squad': ['511533', '169529', '29338...   
378  {'Home': {'Squad': ['168351', '165065', '16928...   
379  {'Home': {'Squad': ['291584', '171788', '16103...   
380  {'Home': {'Squad': ['155703', '176512', '30325...   

                                                     2  \
1    {'Home': {'Squad': ['141698', '137644', '14474...   
2    {'Home': {'Squad': ['137990', '172996', '16795...   
3    {'Home': {'Squad': ['107639', '161940', '45966...   
4    {'Home': {'Squad': ['142720', '185679', '16194...   
5    {'Home': {'Squad': ['293428', '322380', '18260...   
..                                                 ...   
376  {'Home': {'Squad': ['180213', '174360', '15728...   
377  {'Home': {'Squad': ['511533', '169529', '29338...   
378  {'Home': {'Squad': ['168351', '165065', '16928...   
379  {'Home': {'Squad': ['291584', '171788', '16103...   
380  {'Home': {'Squad': ['155703', '176512', '30325...   

                                                     3  \
1    {'Home': {'Squad': ['141698', '137644', '14474...   
2    {'Home': {'Squad': ['137990', '172996', '16795...   
3    {'Home': {'Squad': ['107639', '161940', '31664...   
4    {'Home': {'Squad': ['142720', '185679', '33783...   
5    {'Home': {'Squad': ['293428', '322380', '18260...   
..                                                 ...   
376  {'Home': {'Squad': ['180213', '174360', '15728...   
377  {'Home': {'Squad': ['511533', '169529', '29338...   
378  {'Home': {'Squad': ['168351', '165065', '16928...   
379  {'Home': {'Squad': ['291584', '171788', '16103...   
380  {'Home': {'Squad': ['155703', '176512', '30325...   

                                                     4  \
1    {'Home': {'Squad': ['141698', '137644', '14474...   
2    {'Home': {'Squad': ['137990', '172996', '16795...   
3                                                  NaN   
4    {'Home': {'Squad': ['142720', '185679', '33783...   
5    {'Home': {'Squad': ['293428', '322380', '18260...   
..                                                 ...   
376  {'Home': {'Squad': ['180213', '174360', '15728...   
377  {'Home': {'Squad': ['511533', '169529', '29338...   
378  {'Home': {'Squad': ['168351', '165065', '16928...   
379  {'Home': {'Squad': ['291584', '171788', '16103...   
380  {'Home': {'Squad': ['155703', '176512', '30325...   

                                                     5  \
1    {'Home': {'Squad': ['141698', '137644', '14474...   
2    {'Home': {'Squad': ['137990', '172996', '16795...   
3                                                  NaN   
4    {'Home': {'Squad': ['142720', '185679', '33783...   
5    {'Home': {'Squad': ['293428', '322380', '18260...   
..                                                 ...   
376  {'Home': {'Squad': ['180213', '174360', '15728...   
377  {'Home': {'Squad': ['511533', '169529', '29338...   
378  {'Home': {'Squad': ['168351', '165065', '

Vamos pegar as informações dos ID's dos jogadores com as informações dos Squads. Começaremos pelo campeonato da Série A de 2013 até 2024, e nossa intenção é definir um rating inicial de 1500 para cada jogador.

Com isso, conseguiremos calcular um rating de uma partida e atualizar o rating dos jogadores após ela nesse dataframe com os ID's.

In [11]:

def collect_unique_ids(data_folder, file_names, years):
    """
    Função para coletar IDs únicos de jogadores em arquivos JSON.

    :param data_folder: Caminho para o diretório onde os arquivos JSON estão localizados.
    :param file_names: Lista de nomes de arquivos base (ex: ['Serie_A', 'Serie_B', 'Copa_do_Brasil']).
    :param years: Lista ou range de anos dos arquivos a serem processados.
    :return: Lista de IDs únicos coletados de todos os arquivos.
    """

    unique_ids = set()

    def process_dataframe(df):
        for column in df.columns:
            for cell in df[column]:
                if isinstance(cell, dict):  
                    for key in ['Home', 'Away']:
                        team_info = cell.get(key, {})
                        # Adicionar os IDs do 'Squad'
                        squad_ids = team_info.get('Squad', [])
                        unique_ids.update(squad_ids)


    for file_name in file_names:
        for year in years:
            file_path = os.path.join(data_folder, f'{file_name}_{year}_squads.json')
            if os.path.exists(file_path):
                df = pd.read_json(file_path)
                # df_ = df.iloc[:,8]
                # df = pd.DataFrame(df_)
                df.to_excel('df.xlsx')
                process_dataframe(df)
            else:
                print(f'Arquivo não encontrado: {file_path}')


    return list(unique_ids)

data_folder = '../Data/'
#file_names = ['Serie_A', 'Serie_B', 'Serie_C', 'Serie_D', 'CdB']
file_names = ['Serie_A']
years = range(2014, 2015)

unique_player_ids = collect_unique_ids(data_folder, file_names, years)

print(f'Total de IDs únicos coletados: {len(unique_player_ids)}')

Total de IDs únicos coletados: 736


Total de 15448 diferentes jogadores registrados!

Vamos trabalhar com os 736 da série A 2014. Simularemos inicialmente idades e jogos jogados.

In [12]:
df_players = pd.DataFrame(unique_player_ids, columns=['player'])

# Adicionar a coluna 'rating' com valor inicial de 1500 para cada jogador
df_players['rating'] = 1500
df_players['age'] = 25
df_players['games_played'] = 50
df_players['nome_jogador'] = np.nan

df_players.to_excel('id_players.xlsx')

# Mostrar o DataFrame criado
df_players

player  rating  age  games_played  nome_jogador
0    149237    1500   25            50           NaN
1    331937    1500   25            50           NaN
2    176729    1500   25            50           NaN
3    308980    1500   25            50           NaN
4    171547    1500   25            50           NaN
..      ...     ...  ...           ...           ...
731  185679    1500   25            50           NaN
732  291380    1500   25            50           NaN
733  145298    1500   25            50           NaN
734  174348    1500   25            50           NaN
735  309644    1500   25            50           NaN

[736 rows x 5 columns]